### PR TITLE
fix(VField): stopPropagation mousedown event on clearable icon

### LINKED ISSUE #16925: [Bug Report][3.1.10] Vue 3 [v-select, v-combobox, v-autocomplete]: open-on-clear
### Environment
**Vuetify Version:** 3.1.10
**Last working version:** 3.1.10
**Vue Version:** 3.3.0-alpha.4
**Browsers:** Chrome 111.0.0.0
**OS:** Windows 10

### Steps to reproduce
use clearable prop with v-select/v-combobox/v-autocomplete

### Expected Behavior
clerable icon capture click event

### Actual Behavior
Component options opened

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNq9Vtty2jAQ/RWNXhJmfIEAaeqBTvIdmAdhr4uLLGkkmYRh+PeubEOMSdqmXF4AaS/nnPUZvLMtfVEqWJdAIzqxUCjOLPyIBSGTtc+Uqn5Wh4LlgiScGTONqWL+E3n1x/2YNhlVTsJ0ShLJpXY5Oi+Y3sT0UJX6GYc34j58zCoLgT0GfWxCbG45YEokFQhfCj/hwPQ0hYyV3FYoLRxWWplIJAsW9teERDkKcDjVd0zfI8s8BT8Fy3Lu4q68HUZxMgXeVM7683YwqpiwRUXP6hLaQc4WVd0Rofd42B6OAQ6JvTzdwdfpNlQOkSOeqGMhF/Lt8kwfvs70QOaU6ySsDXfgvr/v+NCUSQIGaZ/hw5pee0rXteDwYhYkHSUnDW7i0dH5Hv2rkhuZeHwJE/+7mM9cfmJz0Frqs0yeMW5u6vLHK7q8EXNbm3+7is07Um7k86cr+fwzNR2ju6NbOepdBA/1MjIJW0sKHk2ic2WJAVu6cF4oqS3ZEg0Z2ZFMy4Lc4XZz55ITKYwlThyZuoT7WRAEL1qzzf33XrCCjbnvzXvtPLNPHHgP3tAbeWPv0WVMwhoXEalHa1Bkq4JfRgrcpLaONM6xCuAziEh14+6QjDvHdGmtMlEYlkKtfgY4uvAZY6Euhc0LfE6yeB4Gw6DvM66WLBiFaW5sOxyAKfyFlq8GNALH1Gth2DzDzetjnCaM7QfBoF/3be58fJTGNT5pGCLKGrSvQaSgAf9p/iCik/uRkE7KiRgHvYvFDufLc7EyndEm7oUekdme4X+orFq4+rmD2Xn0iCXi1pV0/htPM2bN)

### PR DESCRIPTION
**Pull request by yuwu9145**
fixes #16925

## Description

fixes #16925
## Markup:

```vue
<template>
  <v-app>
    <v-main class="pa-8 w-50">
      <v-card color="primary" class="d-flex flex-column w-100" title=":open-on-clear=default">

        <v-autocomplete :items="items" hide-details="auto" v-model="item[0]" :clearable="true" label="autocomplete" />
        <v-select :items="items" hide-details="auto" v-model="item[1]" :clearable="true" label="select" />
        <v-combobox :items="items" hide-details="auto" v-model="item[2]" :clearable="true" label="combobox" />
      </v-card>

      <v-card color="success" class="d-flex flex-column w-100" title=":open-on-clear=true">
        <v-autocomplete :items="items" hide-details="auto" v-model="item[3]" :clearable="true" label="autocomplete"
          :open-on-clear="true" />
        <v-select :items="items" hide-details="auto" v-model="item[4]" :clearable="true" label="select"
          :open-on-clear="true" />
        <v-combobox :items="items" hide-details="auto" v-model="item[5]" :clearable="true" label="combobox"
          :open-on-clear="true" />
      </v-card>

      <v-card color="error" class="d-flex flex-column w-100" title=":open-on-clear=false">
        <v-autocomplete :items="items" hide-details="auto" v-model="item[6]" :clearable="true" label="autocomplete"
          :open-on-clear="false" />
        <v-select :items="items" hide-details="auto" v-model="item[7]" :clearable="true" label="select"
          :open-on-clear="false" />
        <v-combobox :items="items" hide-details="auto" v-model="item[8]" :clearable="true" label="combobox"
          :open-on-clear="false" />
      </v-card>
    </v-main>
  </v-app>
</template>

<script setup>
import { ref } from 'vue'

const item = ref([...Array(9).keys()])
const items = ref([1, 2, 3, 4, 5, 6])
</script>


```

### CODE REVIEW

File: packages/vuetify/src/components/VField/VField.tsx [hunk 1]
```diff
@@ -291,6 +291,10 @@ export const VField = genericComponent<new <T>() => {
               <div
                 class="v-field__clearable"
                 v-show={ props.dirty }
+                onMousedown={ (e: MouseEvent) => {
+                  e.preventDefault()
```
[the remark this answers is not in the document]
Comment (KaelWD) [reply packages/vuetify/src/components/VField/VField.tsx:295]: https://codepen.io/kaelwd/pen/KKxGVaG

[the remark this answers is not in the document]
Comment (yuwu9145) [reply packages/vuetify/src/components/VField/VField.tsx:295]: onMousedown is registered on the div and `stopPropagation` stops it bubbling up to its parent VField. 

My understanding is when mousedown clearable icon with `stopPropagation`, it does not touch VField at all

So logic here has nothing to do with the input?

[the remark this answers is not in the document]
Comment (KaelWD) [reply packages/vuetify/src/components/VField/VField.tsx:295]: mousedown.prevent on an element will prevent blurring *any* input when that element is clicked

**Comment by KaelWD** [comment]
VTextField switched to separate prop/event since then (#15907) so it should be fine to revert #15025

File: packages/vuetify/src/components/VField/VField.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnleider) [reply packages/vuetify/src/components/VField/VField.tsx:295]: This is a video without preventDefault and then with:

preventDefault

File: packages/vuetify/src/components/VCombobox/VCombobox.tsx [hunk 2]
```diff
@@ -119,6 +120,7 @@ export const VCombobox = genericComponent<new <
       },
     })
     const selectionIndex = ref(-1)
+    const cleared = ref(false)
```
[the remark this answers is not in the document]
Comment (johnleider) [reply packages/vuetify/src/components/VCombobox/VCombobox.tsx:123]: Removing the ref breaks the fix.

**Review by johnleider — DISMISSED** [review]

File: packages/vuetify/src/components/VCombobox/VCombobox.tsx [hunk 3]
```diff
@@ -156,12 +157,20 @@ export const VCombobox = genericComponent<new <
...
 
         if (!val) selectionIndex.value = -1
-        if (isFocused.value) menu.value = true
 
         isPristine.value = !val
       },
     })
     watch(_search, value => {
+      if (cleared.value) {
+        // Implement open-on-clear (https://github.com/vuetifyjs/vuetify/issues/16925)
```
Comment (yuwu9145) [inline packages/vuetify/src/components/VCombobox/VCombobox.tsx:166]: Just add a single comment to better record its original purpose for future reference

Comment (johnleider) [reply packages/vuetify/src/components/VCombobox/VCombobox.tsx:166]: This is what git is for. We typically only put links here if it references an outside resource.

**Review by johnleider — APPROVED** [review]

**Comment by yuwu9145** [comment]
@johnleider 

I tried/experimented my old solution of using `onInput={ onInput }` again today.

Since Kael mentioned https://github.com/vuetifyjs/vuetify/pull/15907, it means `onInput={ onInput }` on Autocomplete should become unnecessary/redundant. Letting `v-model` drive `search.value` should be the right direction to go.

Probably we should try to refactor that `onInput={ onInput }` on Autocomplete sometime later


For this issue, I gave this PR an "approve"

**Pull request merged**